### ISSUE TITLE
colab error

### ISSUE THREAD

**Issue by DrJesseHansen** [dptech-corp/Uni-Fold#116]
I encountered the error below when running through your google colab.

fused_multi_tensor is not installed corrected
fused_rounding is not installed corrected
fused_softmax is not installed corrected
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
[<ipython-input-3-380200f791ad>](https://localhost:8080/#) in <cell line: 8>()
      6 import numpy as np
      7 from pathlib import Path
----> 8 from unifold.colab.data import validate_input
      9 from unifold.msa.utils import divide_multi_chains
     10 

5 frames
[/usr/local/lib/python3.9/dist-packages/unifold/symmetry/utils/msym_utils.py](https://localhost:8080/#) in <module>
      1 from functools import reduce
      2 import string
----> 3 from libmsym import libmsym as m
      4 import numpy as np
      5 from typing import *

ModuleNotFoundError: No module named 'libmsym'

---------------------------------------------------------------------------
NOTE: If your import is failing due to a missing package, you can
manually install dependencies using either !pip or !apt.

To view examples of installing some common dependencies, click the
"Open Examples" button below.
---------------------------------------------------------------------------

**Comment by ZiyaoLi** [comment]
this should be fixed in #116.

before merging, you may refer to notebook on branch `Ziyao-patch-2`.

**Comment by DrJesseHansen** [comment]
thanks.  Not sure exactly what you mean here but I tried the following:
https://colab.research.google.com/github/googlecolab/colabtools/blob/master
from there I put in: https://github.com/dptech-corp/Uni-Fold/tree/ZiyaoLi-patch-2
then I clicked unifold notebook.   It loaded, I tried again, and it failed.   

```
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
[<ipython-input-6-380200f791ad>](https://localhost:8080/#) in <cell line: 8>()
      6 import numpy as np
      7 from pathlib import Path
----> 8 from unifold.colab.data import validate_input
      9 from unifold.msa.utils import divide_multi_chains
     10 

5 frames
[/usr/local/lib/python3.9/dist-packages/unifold/symmetry/utils/msym_utils.py](https://localhost:8080/#) in <module>
      1 from functools import reduce
      2 import string
----> 3 from libmsym import libmsym as m
      4 import numpy as np
      5 from typing import *

ModuleNotFoundError: No module named 'libmsym'

---------------------------------------------------------------------------
```


therefore I added a cell above "generate MSAs" with the following command.  

`pip install libmsym
`
Then it runs now.  thanks.

**Comment by DrJesseHansen** [comment]
thanks.  I ran my multimer job (6 copies) an the output showed no interaction at all.   Like it's just 6 copies of my protein very spread out in a 6fold positions.   Yet I know that my protein SHOULD interact as 6 copies.  Is there any way to push the oligomerization a bit harder?  thanks.

**Comment by vprobon** [comment]
When trying to run the colab notebook I get the error at the bottom of this message.
This happens using the two different branches available:
- https://colab.research.google.com/github/dptech-corp/Uni-Fold/blob/main/notebooks/unifold.ipynb
- https://colab.research.google.com/github/dptech-corp/Uni-Fold/blob/ZiyaoLi-patch-2/notebooks/unifold.ipynb#

It seems there is a compatibility issue with the pytorch (currently torch 2.0.0 in Google colab). 
I tried within colab to uninstall torch 2.0.0 and install 1.13.1 (which should be compatible with Uni-Core) but there are 

I also tried to install the newest Uni-Core library
!wget https://github.com/dptech-corp/Uni-Core/releases/download/0.0.2/unicore-0.0.1+cu118torch2.0.0-cp39-cp39-linux_x86_64.whl
!pip3 -q install "unicore-0.0.1+cu118torch2.0.0-cp39-cp39-linux_x86_64.whl"

but still it does not work. 

Could you please suggest a workaround?

###

---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
[<ipython-input-3-380200f791ad>](https://localhost:8080/#) in <cell line: 8>()
      6 import numpy as np
      7 from pathlib import Path
----> 8 from unifold.colab.data import validate_input
      9 from unifold.msa.utils import divide_multi_chains
     10 

4 frames
[/usr/local/lib/python3.10/dist-packages/unifold/__init__.py](https://localhost:8080/#) in <module>
      3 import argparse
      4 
----> 5 from . import task, model, loss

[/usr/local/lib/python3.10/dist-packages/unifold/task.py](https://localhost:8080/#) in <module>
      7 import numpy as np
      8 
----> 9 from unifold.dataset import UnifoldDataset, UnifoldMultimerDataset
     10 from unicore.data import data_utils
     11 from unicore.tasks import UnicoreTask, register_task

[/usr/local/lib/python3.10/dist-packages/unifold/dataset.py](https://localhost:8080/#) in <module>
      6 import torch
      7 from typing import *
----> 8 from unifold.data import utils
      9 from unifold.data.data_ops import NumpyDict, TorchDict
     10 from unifold.data.process import process_features, process_labels

[/usr/local/lib/python3.10/dist-packages/unifold/data/utils.py](https://localhost:8080/#) in <module>
      8 from typing import *
      9 
---> 10 from . import residue_constants as rc
     11 from .data_ops import NumpyDict
     12 

[/usr/local/lib/python3.10/dist-packages/unifold/data/residue_constants.py](https://localhost:8080/#) in <module>
     21 
     22 import numpy as np
---> 23 from unicore.utils import tree_map
     24 
     25 # Distance from one CA to next CA [trans configuration: omega = 180].

ModuleNotFoundError: No module named 'unicore'


####

**Comment by burunduk57** [comment]
I have the same problem:
ModuleNotFoundError: No module named 'unicore'

**Comment by ZiyaoLi** [comment]
All problems including 

```
ModuleNotFoundError: No module named 'unicore'
```

are due to inconsistent versions of unicore wheel and colab environments.

I notice that colab starts to update their python and torch versions much more frequent than before. The latest fix is in #118 which update the target wheel to cp310+torch2.0.0.

If this error raises again, users can either

- select proper wheel version from [https://github.com/dptech-corp/Uni-Core/releases] (will be updated occasionally), and change the UNICORE_URL path in the 2nd block in the notebook, as well as the downloaded file path.

or

- git clone and pip install Uni-Core from https://github.com/dptech-corp/Uni-Core . This takes longer.

**Comment by vprobon** [comment]
Thank you @ZiyaoLi for the update!
Best
V
